xdsl/transforms/convert_pdl_to_pdl_interp/conversion.py

```diff
@@ -27,3 +37,35 @@ def detect_roots(self, pattern: pdl.PatternOp) -> list[OpResult[pdl.OperationTyp
             if isinstance(op, pdl.OperationOp) and op.op not in used
         ]
         return roots
+
+    def _extract_type_predicates(
+        self,
+        type_value: Operation | SSAValue,
```

Comment:
I'm a little suspicious of this, why is it better for it to be passed in as either a value or an op like this rather than just the op, letting the caller resolve the value to the op if needed?